oak_runtime/src/node/http/client.rs

```diff
@@ -0,0 +1,327 @@
+                    "Failed to read request",
+                );
+                return;
+            }
+        };
+        debug!("Incoming HTTP request: {:?}", request);
+
+        let mut hyper_request_builder = hyper::Request::builder()
+            .method(&request.method[..])
+            .uri(request.uri);
+
+        if let Some(headers) = request.headers {
+            for (header_name, header_value) in headers.into_iter() {
+                hyper_request_builder = hyper_request_builder.header(header_name, header_value);
+            }
+        }
+        let hyper_request = match hyper_request_builder.body(hyper::Body::from(request.body)) {
+            Ok(request) => request,
+            Err(err) => {
+                send_error(
+                    format!("{:?}", err),
+                    http::StatusCode::INTERNAL_SERVER_ERROR,
+                    "Failed to convert the request body",
+                );
+                return;
+            }
+        };
+
+        // Check that the `host` in the request URI matches the authority supported by the node, if
+        // any.
+        if let Some(authority) = self.authority.clone() {
+            if let Some(req_authority) = hyper_request.uri().authority() {
+                if *req_authority != authority {
+                    send_error(
+                        format!("{:?}", req_authority),
+                        http::StatusCode::UNAUTHORIZED,
+                        "Invalid authority",
+                    );
+                    return;
+                }
+            }
+        }
+
+        // Forward the request to the external HTTP service and wait for the response.
+        let resp = match self.http_client.request(hyper_request).await {
+            Ok(rsp) => rsp,
+            Err(err) => {
+                send_error(
+                    format!("{:?}", err),
+                    http::StatusCode::INTERNAL_SERVER_ERROR,
+                    "Error while waiting for the response",
+                );
+                return;
+            }
+        };
+
+        let mut response_handler = ResponseHandler::new(runtime.clone(), resp, invocation);
+        if let Err(err) = response_handler.handle().await {
+            error!("Could not send the response: {}", err);
+        }
+    }
+}
+
+struct ResponseHandler<'a> {
+    runtime: RuntimeProxy,
+    response: http::Response<hyper::Body>,
+    invocation: &'a Invocation,
+}
+
+impl<'a> ResponseHandler<'a> {
+    fn new(
+        runtime: RuntimeProxy,
+        response: http::Response<hyper::Body>,
+        invocation: &'a Invocation,
+    ) -> Self {
+        ResponseHandler {
+            runtime,
+            response,
+            invocation,
+        }
+    }
+
+    async fn handle(&mut self) -> Result<(), OakError> {
+        let headers = Some(HeaderMap::from(self.response.headers().to_owned()));
+        let status = self.response.status().as_u16() as i32;
```

Comment:
Why do we need it a signed?

Reply:
To match the type in the `prost`-generated type (`HttpResponse`).